falcon/asgi/app.py

```diff
@@ -359,8 +359,21 @@ async def __call__(self, scope, receive, send):  # noqa: C901
                 f'The ASGI http scope version {spec_version} is not supported.'
             )
 
+        # NOTE(kgriffs): Per the ASGI spec, we should not proceed with request
+        #   processing until after we receive an initial 'http.request' event.
+        first_event = await receive()
```

Comment:
Presumably the ASGI server could also cancel the task, resulting in a raised exception here.